icu4c/source/common/charstr.cpp

```diff
@@ -197,7 +197,14 @@ CharString &CharString::appendPathPart(StringPiece s, UErrorCode &errorCode) {
     }
     char c;
     if(len>0 && (c=buffer[len-1])!=U_FILE_SEP_CHAR && c!=U_FILE_ALT_SEP_CHAR) {
-        append(U_FILE_SEP_CHAR, errorCode);
+        char dirSepChar = U_FILE_SEP_CHAR;
+#if (U_FILE_SEP_CHAR != U_FILE_ALT_SEP_CHAR)
```

Comment:
optional: i would remove the parentheses

Reply:
FWIW, I think most of the other places that use/check for `(U_FILE_SEP_CHAR != U_FILE_ALT_SEP_CHAR)` do have parentheses though.